contrib/enfuse_pro_2.lua

```diff
@@ -0,0 +1,1050 @@
+	dt.print_log("Completed Blend")
+
+	return resultBlend, path_with_filename
+end
+local function show_status(storage, image, format, filename, --Store: Called on each exported image
+  number, total, high_quality, extra_data)
+     dt.print(_('export TIFF for image fusion ')..tostring(truncate(number)).." / "..tostring(truncate(total)))   
+end
+function create_image_fusion(storage, image_table, extra_data) --Finalize: Called once when all images are done exporting	
+--Create Images String--
+	images_to_align = ""
+	images_to_blend = ""
+	counted_images=0
+	first_id = "999999999999999999999999"
+	last_id = "0"
+	for source_image,image_path in pairs(image_table) do
+		counted_images=counted_images+1
+		images_to_align = images_to_align..image_path.." "
+		curr_path, curr_filename, curr_id, curr_ext = GetFileName(image_path)
+		if (curr_id < first_id) then 
+			first_path = curr_path
+			first_filename = curr_filename
+			first_id = curr_id
+			first_ext = curr_ext
+			dt_source_image = source_image
+		end 
+		if (curr_id > last_id) then 
+			last_path = curr_path
+			last_filename = curr_filename
+			last_id = curr_id
+			last_ext = curr_ext
+		end
+		source_path = source_image.path
+		source_file = source_image.path..os_path_seperator..source_image.filename
+	end
+
+--Check if at least 2 images selected--
+	if (counted_images<=1) then
+		dt.print(_('ERROR: not enough pictures selected. please select two or more images\nfrom the same object, but with different camera settings.'))
+		dt.print_error("Not enough pictures selected")
+		remove_temp_files()
+		return
+	end
+
+--Ensure Required Software is Installed--
+	df.set_executable_path_preference("align_image_stack", dt.preferences.read("module_enfuse_pro", "ais_path", "string"))
+	dt.print_log("Executable Path Preference: "..df.get_executable_path_preference("align_image_stack"))
+	AIS_path = df.check_if_bin_exists("align_image_stack")
+	if not AIS_path then
+		dt.print_error("align image stack not found")
+		dt.print("ERROR - align image stack not found")
+		remove_temp_files()
+		return
+	end
+	
+	df.set_executable_path_preference("enfuse", dt.preferences.read("module_enfuse_pro", "enfuse_path", "string"))
+	dt.print_log("Executable Path Preference: "..df.get_executable_path_preference("enfuse"))
+	enfuse_path = df.check_if_bin_exists("enfuse")
+	if not enfuse_path then
+		dt.print_error("enfuse not found")
+		dt.print("ERROR - enfuse not found")
+		remove_temp_files()
+		return
+	end
+	
+	df.set_executable_path_preference("exiftool", dt.preferences.read("module_enfuse_pro", "exiftool_path", "string"))
+	dt.print_log("Executable Path Preference: "..df.get_executable_path_preference("exiftool"))
+	exiftool_path = df.check_if_bin_exists("exiftool")
+	if not exiftool_path then
+		dt.print_error("exiftool not found")
+		dt.print("ERROR - exiftool not found")
+		remove_temp_files()
+		return
+	end
+	
+--Ensure proper version of Enfuse installed--
+	--[[
```

Comment:
This script requires enfuse 4.2.  If you try running it against 4.1 it crashes on the options.  If you're running CentOS, or any Ubuntu LTS except the latest one, then you have version 4.1 by default.  To get around windows not having a grep command you could use find, or you could run enfuse --version in an io.popen() and grab the result, then loop through the lines and use string.match to check for enfuse 4.2